File: src/stores/cart/cartStore.java
import { ref, computed } from 'vue';
import { defineStore } from 'pinia';
import axios from 'axios';

export const useCartStore = defineStore('cart', () => {
  
  const items = ref([]);

 
  const totalQuantity = computed(() =>
    items.value.reduce((total, item) => total + item.quantity, 0)
  );

  
  const totalPrice = computed(() =>
    items.value.reduce((total, item) => total + item.price * item.quantity, 0)
  );

 
  const addItem = (product, quantity = 1) => {
    const existingItem = items.value.find(item => item.id === product.id);
    if (existingItem) {
      existingItem.quantity += quantity; 
    } else {
      items.value.push({ ...product, quantity }); 
    }
  };

  const removeItem = (id) => {
    items.value = items.value.filter(item => item.id !== id);
  };

  
  const updateQuantity = (id, quantity) => {
    const item = items.value.find(item => item.id === id);
    if (item) {
      if (quantity > 0) {
        item.quantity = quantity;
      } else {
        removeItem(id); 
      }
    }
  };

  const clearCart = () => {
    items.value = [];
  };

  return {
    items,
    totalQuantity,
    totalPrice,
    addItem,
    removeItem,
    updateQuantity,
    clearCart
  };
});

File: src/stores/cart/cartStore.js
import { ref, computed, watch } from 'vue';
import { defineStore } from 'pinia';

export const useCartStore = defineStore('cart', () => {
  const products = ref(JSON.parse(localStorage.getItem('cart') || '[]'));

  const totalQuantity = computed(() => {
    return products.value.reduce((total, product) => total + product.quantity, 0);
  });

  const totalPrice = computed(() => {
    const total = products.value.reduce((sum, product) => sum + product.price * product.quantity, 0);
    return Math.round(total * 100) / 100;
  });

  const addProduct = (product, quantity = 1) => {
    console.log('Producto a añadir:', product);
    const existingProduct = products.value.find(item => item.id === product.id);
    if (existingProduct) {
      existingProduct.quantity += quantity;
    } else {
      products.value.push({ ...product, quantity, imageHash: product.imageHash });
    }
    saveCart();
  };

  const removeProduct = (id, quantity = 1) => {
    const product = products.value.find(product => product.id === id);
    if (product) {
      if (product.quantity > quantity) {
        product.quantity -= quantity;
      } else {
        products.value = products.value.filter(product => product.id !== id);
      }
      saveCart();
    }
  };

  const updateQuantity = (id, quantity) => {
    const product = products.value.find(product => product.id === id);
    if (product) {
      if (quantity > 0) {
        product.quantity = quantity;
      } else {
        removeProduct(id);
      }
    }
    saveCart();
  };

  const clearCart = () => {
    products.value = [];
    saveCart();
  };

  const saveCart = () => {
    localStorage.setItem('cart', JSON.stringify(products.value));
  };

  watch(products, saveCart, { deep: true });

  return {
    products,       
    totalQuantity,  
    totalPrice,    
    addProduct,     
    removeProduct,  
    updateQuantity, 
    clearCart      
  };
});
